proto/substrait/algebra.proto

```diff
@@ -267,11 +267,23 @@ message CrossRel {
 message FetchRel {
   RelCommon common = 1;
   Rel input = 2;
-  // the offset expressed in number of records
-  int64 offset = 3;
-  // the amount of records to return
-  // use -1 to signal that ALL records should be returned
-  int64 count = 4;
+  oneof offset_mode {
+    // the offset expressed in number of records
+    // Deprecated: use `offset_expr` instead
+    int64 offset = 3 [deprecated = true];
+    // Expression evaluated into an integer specifying the number of records to
+    // skip.
+    Expression offset_expr = 5;
```

Comment:
Based on the discussion at the meeting this week we should likely have some sort of recommendation for the expected type here.  Obviously if a consumer can't handle the type (it comes back as a string) it should not accept the plan but we should also recommend a type.

Reply:
Thanks, @EpsilonPrime . Just a bit of guidance since I didn't join the meeting: Should the recommendation be added here as a comment? Should it also go into the spec documentation? And should we proceed with `i64` as the recommendation, given that it was originally used for `offset` and `count`?

Reply:
I have made the corresponding changes and pushed a new commit.

Reply:
I think the update you made is sufficient for this patch. We can address the broader issue in that other ticket.